src/Amazon.Common.DotNetCli.Tools/Utilities.cs

```diff
@@ -195,14 +195,78 @@ public static string LookupTargetFrameworkFromProjectFile(string projectLocation
             return element?.Value;
         }
 
-        public static string LookupOutputTypeFromProjectFile(string projectLocation)
+        /// <summary>
+        /// Uses `dotnet msbuild` to determine project properties.
+        /// </summary>
+        /// <param name="projectLocation">Path of the project</param>
+        /// <param name="propertyName">The name of the property</param>
+        /// <returns>The value of the project property</returns>
+        private static string LookupMsbuildProperty(string projectLocation, string propertyName)
         {
-            var projectFile = FindProjectFileInDirectory(projectLocation);
+            var process = new Process();
+            string outputJson;
+            try
+            {
+                process.StartInfo = new ProcessStartInfo()
+                {
+                    FileName = "dotnet",
+                    Arguments = $"msbuild {projectLocation} -getProperty:{propertyName}",
```

Comment:
Possible alternative approach using `Microsoft.Build` and `Micorosft.Build.Locator`.

I had explored this for Lambda Annotations, but couldn't use that there because source generators have to target .NET Standard and these packages don't. 

I haven't explored all the tradeoffs, but might be nice to not have to manage our own `Process` and the .NET 8+ issue Norm mentioned. There are some implications around Visual Studio compatibility when selecting versions of these dependencies.

```
using Microsoft.Build.Evaluation;
using Microsoft.Build.Locator;

var instance = MSBuildLocator.QueryVisualStudioInstances().OrderByDescending(
       instance => instance.Version).First(); // select which of the installed instances to use

// Register a specific instance of MSBuild
MSBuildLocator.RegisterInstance(instance);

// You have to register in a different method than when you start consuming it.
DetectProjectType("path\\to\\project.csproj");

static void DetectProjectType(string pathToCsproj)
{
    var project = new Project(pathToCsproj);

    var outputType = project.GetProperty("OutputType");

    Console.WriteLine(outputType.EvaluatedValue);
}
```

Reply:
This method makes me nervous of something not working write on mac and linux platforms. The locator seems a little flaky for a component automatically picked up in people's CI systems.